### PR TITLE
Added :meth:`~.Mobject.match_points`

### PR DESCRIPTION
**Pull request by Darylgolden**
## Changelog / Overview

- Added :func:`~.Mobject.become_points`, which transforms the points, positions and submobjects of a Mobject to match that of the other while keeping style unchanged.
- Added example in docs.
- Added test.

## Motivation
While working on creating polyhedra on manim, I found transforming polygons while keeping the style (which could be set by the user after initialization) unchanged difficult. 

## Explanation for Changes
Example:
```py
class BecomePoints(Scene):
    def construct(self):
        a = Triangle(fill_color=YELLOW, fill_opacity=0.8)
        b = Square(fill_color=GREEN, fill_opacity=0.2)
        self.add(a)
        self.play(a.animate.become_points(b))
        self.wait()
```
generates:

https://user-images.githubusercontent.com/9102529/115170599-95767e80-a0f3-11eb-86c3-c2eb9c3c6237.mp4

## Testing Status
Added test in test_transform. Added control data.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by Darylgolden** [comment]
The documentation scenes render incorrectly, but they render correctly on my system. I encountered a similar issue where manim didn't recognize the new functions I made until I restarted poetry shell. Perhaps something similar is going on here?

**Comment by Darylgolden** [comment]
The documentation fails to render correctly locally too. But given that the tests pass in the CI, and the control data generated is correct, the git repo does have the correct changes. Not sure why sphinx doesn't render it properly.

**Comment by abhi3019** [comment]
Looks good, thanks! Regarding the method `become_style` - there is already a `match_style` method that does the same job, but for `VMobject`s though. https://github.com/ManimCommunity/manim/blob/0909eeb556a869f30b1cb8f85bbc89bb9337c0a4/manim/mobject/types/vectorized_mobject.py#L278-L291
If you meant to generalize it to `Mobject`s, the `interpolate_color` method you've used [here ](https://github.com/ManimCommunity/manim/pull/1364/files#diff-fa8fe466fb7275e64543789145cbd2e35b267715610a6376591ddbda0d2f6194R2331)is implemented for `VMobject`s but not for `Mobject`s
https://github.com/ManimCommunity/manim/blob/0909eeb556a869f30b1cb8f85bbc89bb9337c0a4/manim/mobject/mobject.py#L2261-L2262 https://github.com/ManimCommunity/manim/blob/0909eeb556a869f30b1cb8f85bbc89bb9337c0a4/manim/mobject/types/vectorized_mobject.py#L1241
So the method `become_style` looks redundant to me. Also, I'd prefer `match_points` over `become_points`

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -2287,6 +2287,28 @@ def construct(self):
...
+        --------
+        .. manim:: BecomeScene
+
+            class MatchPointsScene(Scene):
+                def construct(self):
+                    circ = Circle(fill_color=RED, fill_opacity=0.8)
+                    square = Square(fill_color=BLUE, fill_opacity=0.2)
+                    self.add(circ)
+                    self.wait(0.5)
+                    circ.match_points(square)
```
Comment (abhi3019) [inline manim/mobject/mobject.py:2304]: Wouldn't it be better if we actually show the transformation?
```suggestion
                    self.play(circ.animate.match_points(square))
```

File: tests/test_graphical_units/test_transform.py [hunk 2]
```diff
@@ -132,6 +132,15 @@ def construct(self):
         self.play(FadeOut(square))
 
 
+class MatchPointsScene(Scene):
+    def construct(self):
+        circ = Circle(fill_color=RED, fill_opacity=0.8)
+        square = Square(fill_color=BLUE, fill_opacity=0.2)
+        self.add(circ)
+        circ.match_points(square)
+        self.wait()
```
Comment (abhi3019) [inline tests/test_graphical_units/test_transform.py:141]: Just to be consistent with other tests
```suggestion
        self.play(circ.animate.match_points(square))
```

File: manim/mobject/mobject.py (hunk 1 above)
Comment (Darylgolden) [reply manim/mobject/mobject.py:2304]: I was trying to match the `become` example, but I agree that this is better.

**Comment by abhi3019** [comment]
The title should be changed to - Added :func:`match_points` or probably to - Added :meth:`~Mobject.match_points`, otherwise LGTM, thanks!

**Review by abhi3019 — APPROVED** [review]

**Comment by Darylgolden** [comment]
The docs are still not rendering properly, and I have no idea why.

**Comment by Darylgolden** [comment]
https://manimce--1364.org.readthedocs.build/en/1364/reference/manim.mobject.mobject.Mobject.html?highlight=match_points#manim.mobject.mobject.Mobject.match_points It seems like a weird caching issue, since it doesn't even wait the one second self.play should make it wait.

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -2287,6 +2287,28 @@ def construct(self):
             sm1.interpolate_color(sm1, sm2, 1)
         return self
 
+    def match_points(self, mobject: "Mobject", copy_submobjects: bool = True):
+        """Edit points, positions, and submobjects to be identical
+        to another :class:`~.Mobject`, while keeping the style unchanged.
+
+        Examples
+        --------
+        .. manim:: BecomeScene
```
Comment (abhi3019) [inline manim/mobject/mobject.py:2296]: Ah, I think I found the issue. This should probably be `MatchPointsScene`, not `BecomeScene`
```suggestion
        .. manim:: MatchPointsScene
```

Comment (Darylgolden) [reply manim/mobject/mobject.py:2296]: Wow, can't believe I didn't spot that. Thanks!

**Review by abhi3019 — APPROVED** [review]

**Review by huguesdevimeux — APPROVED** [review]
SGTM

**Review by jsonvillanueva — APPROVED** [review]

**Pull request merged**